core/src/main/java/org/apache/iceberg/rest/ScanTaskIterable.java

```diff
@@ -201,20 +198,21 @@ public boolean hasNext() {
         return true;
       }
 
-      while (true) {
+      while (!shutdown.get()) {
```

Comment:
I think this more gracefully handles a close() call. If someone calls close(), shutdown will be set and the tasks will be cleared. If the consumer thread is already on line 203, it'll get null and then loop back to shutdown.get() which will be true and then exit, returning false on 215.